core/src/test/java/org/elasticsearch/action/bulk/BulkItemResponseTests.java

```
@@ -19,17 +19,149 @@
 
 public class BulkItemResponseTests extends ESTestCase {
+
     public void testFailureToString() {
         Failure failure = new Failure("index", "type", "id", new RuntimeException("test"));
         String toString = failure.toString();
         assertThat(toString, containsString("\"type\":\"runtime_exception\""));
         assertThat(toString, containsString("\"reason\":\"test\""));
         assertThat(toString, containsString("\"status\":500"));
     }
+
+    public void testToAndFromXContent() throws IOException {
+        final XContentType xContentType = randomFrom(XContentType.values());
+
+        for (DocWriteRequest.OpType opType : DocWriteRequest.OpType.values()) {
+            int bulkItemId = randomIntBetween(0, 100);
+            boolean humanReadable = randomBoolean();
+            BytesReference originalBytes = null;
+            BulkItemResponse expectedBulkItemResponse = null;
+
+            if (opType == DocWriteRequest.OpType.INDEX || opType == DocWriteRequest.OpType.CREATE) {
+                expectedBulkItemResponse = new BulkItemResponse(bulkItemId, opType, randomIndexResponse());
+                originalBytes = toXContent(expectedBulkItemResponse, xContentType, humanReadable);
+
+            } else if (opType == DocWriteRequest.OpType.DELETE) {
+                expectedBulkItemResponse = new BulkItemResponse(bulkItemId, opType, randomDeleteResponse());
+                originalBytes = toXContent(expectedBulkItemResponse, xContentType, humanReadable);
+
+            } else if (opType == DocWriteRequest.OpType.UPDATE) {
+                Tuple<UpdateResponse, UpdateResponse> updates = randomUpdateResponse(xContentType);
+                expectedBulkItemResponse = new BulkItemResponse(bulkItemId, opType, updates.v2());
+                originalBytes = toXContent(new BulkItemResponse(bulkItemId, opType, updates.v1()), xContentType, humanReadable);
+
+            } else {
+                fail("Test does not support opType [" + opType + "]");
```

Is this is necessary given we loop over OpType.values()? If other values are added in the future we should fail because expectedBulkItemResponse/originalBytes is not set anyway.